wm/docker/image/Dockerfile

```diff
@@ -1,77 +1,80 @@
-FROM centos:7
+FROM almalinux:9
 USER root
 
-RUN yum upgrade -y \
-    && yum clean all \
-    && rm -rf /var/cache/yum
+RUN dnf upgrade -y \
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
 
+RUN dnf install -y git which\
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
 
-RUN yum install -y git curl which\
-    && yum clean all \
-    && rm -rf /var/cache/yum
-
-
-RUN yum install -y epel-release.noarch\
-    && yum clean all \
-    && rm -rf /var/cache/yum
+RUN dnf install -y epel-release.noarch\
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
 
 
 # PKI stuff
-RUN yum install -y https://repo.opensciencegrid.org/osg/3.5/osg-3.5-el7-release-latest.rpm\
-    && yum clean all \
-    && rm -rf /var/cache/yum
+RUN dnf install -y https://repo.opensciencegrid.org/osg/3.6/osg-3.6-el9-release-latest.rpm\
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
 
-RUN yum install -y osg-pki-tools\
-    && yum clean all \
-    && rm -rf /var/cache/yum
+RUN dnf install -y osg-pki-tools\
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
 
-#RUN rpm -i http://mirror.grid.uchicago.edu/pub/osg/3.3/el7/testing/x86_64/voms-2.0.14-1.3.osg33.el7.x86_64.rpm
-RUN rpm -i http://mirror.grid.uchicago.edu/pub/osg/3.5/el7/release/x86_64/voms-2.0.14-1.6.osg35.el7.x86_64.rpm\
-    && yum clean all \
-    && rm -rf /var/cache/yum
+RUN dnf install -y voms\
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
 
-RUN rpm -i http://mirror.grid.uchicago.edu/pub/osg/3.5/el7/release/x86_64/voms-clients-cpp-2.0.14-1.6.osg35.el7.x86_64.rpm\
-    && yum clean all \
-    && rm -rf /var/cache/yum
+#RUN rpm -i http://mirror.grid.uchicago.edu/pub/osg/3.5/el7/release/x86_64/voms-clients-cpp-2.0.14-1.6.osg35.el7.x86_64.rpm\
+#    && dnf clean all \
+#    && rm -rf /var/cache/dnf
 
-RUN yum install -y osg-ca-certs\
-    && yum clean all \
-    && rm -rf /var/cache/yum
+RUN dnf install -y voms-clients-cpp \
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
 
+RUN dnf install -y osg-ca-certs\
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
 
 COPY vomses /etc
 
 # Oracle client
-RUN yum install -y libaio\
-    && yum clean all \
-    && rm -rf /var/cache/yum
+RUN dnf install -y libaio\
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
+
+#RUN rpm -i https://download.oracle.com/otn_software/linux/instantclient/2112000/el9/oracle-instantclient6-basic-21.12.0.0.0-1.el9.x86_64.rpm
+#RUN dnf install -y oracle-release-el9
+#RUN dnf install -y oracle-instantclient-release-el9
+#RUN dnf install -y oracle-instantclient-basic
+COPY oracle-instantclient-basic-21.12.0.0.0-1.el9.x86_64.rpm /tmp
+RUN dnf install -y /tmp/oracle-instantclient-basic-21.12.0.0.0-1.el9.x86_64.rpm
```

Comment:
This assumes that we already downloaded the rpm. 

I believe one option is to add the curl command to download the rpm; then install it; and then remove the rpm. 
Its 72 MB; so it should be fine. 

The second option is to add the oracle yum server to the package manger configuration. 
See instructions [here](https://yum.oracle.com/getting-started.html#:~:text=Red%20Hat%20Enterprise%20Linux%209)

Reply:
@ericvaandering any preferences, here?